### PR TITLE
Get rid of StatTypes

### PR DESCRIPTION
**Pull request by bhollis**
This purges Loadout Optimizer of the `StatTypes` stringy stat keys.

### CODE REVIEW

**Comment by ryan-rushton** [comment]
So glad you did this, I have been meaning to for a while.

File: src/app/loadout-builder/generated-sets/GeneratedSetButtons.tsx [hunk 1]
```diff
@@ -46,9 +46,9 @@ export default function GeneratedSetButtons({
 
   const statsWithPlus5: number[] = [];
 
-  for (const [stat, value] of Object.entries(set.stats)) {
+  for (const [statHash, value] of Object.entries(set.stats)) {
     if (value % 10 > 4) {
-      statsWithPlus5.push(statHashes[stat]);
+      statsWithPlus5.push(parseInt(statHash, 10));
```
Comment (ryan-rushton) [inline src/app/loadout-builder/generated-sets/GeneratedSetButtons.tsx:51]: The fact that these hashes are strings in some places and numbers in others is odd. I think that is on the API thought from my very brief investigation.

File: src/app/loadout-builder/process-worker/process.ts [hunk 2]
```diff
@@ -55,42 +60,32 @@ export function process(
...
-  const orderedStatHashes = statOrder.map((statType) => statHashes[statType]);
   // Stat types excluding ignored stats
-  const orderedConsideredStats = statOrder.filter((statType) => !statFilters[statType].ignored);
-  const orderedConsideredStatHashes = orderedConsideredStats.map(
-    (statType) => statHashes[statType]
+  const orderedConsideredStatHashes = statOrder.filter(
+    (statHash) => !statFilters[statHash].ignored
   );
-  const statHashToOrder: { [statHash: number]: number } = {};
-  statOrder.forEach((statType, index) => (statHashToOrder[statHashes[statType]] = index));
```
Comment (ryan-rushton) [inline src/app/loadout-builder/process-worker/process.ts:82]: So good!

**Review by ryan-rushton — APPROVED** [review]
LGTM. I had a bit of a click around and everything seemed to be working right afterwards.

File: src/app/loadout-builder/generated-sets/GeneratedSetButtons.tsx (hunk 1 above)
Comment (bhollis) [reply src/app/loadout-builder/generated-sets/GeneratedSetButtons.tsx:51]: No, they're always numbers. But keys of objects in JavaScript are always strings. We fake it most of the time based on the property that if you use a string in number context it'll turn into a number, but many things are correctly typed in TS as considering object properties as strings. This means sometimes you need to make the conversion explicit. 

Another approach would be to switch to `Map<number, whatever>` which is more typesafe and maaaybe more performant (but also maybe slower) but it's definitely harder to use. Worth an experiment to see if it improves performance at some point.

**Pull request merged**